src/item.h

```diff
@@ -382,6 +383,20 @@ class item : public visitable<item>
     */
     std::string info( bool showtext, std::vector<iteminfo> &dump, int batch ) const;
 
+    /**
+    * Return all the information about the item and its type, and dump to vector.
+    *
+    * This includes the different
+    * properties of the @ref itype (if they are visible to the player). The returned string
+    * is already translated and can be *very* long.
+    * @param parts controls which parts of the iteminfo to return.
+    * @param dump The properties (encapsulated into @ref iteminfo) are added to this vector,
+    * the vector can be used to compare them to properties of another item.
+    * @param batch The batch crafting number to multiply data by
+    */
+    std::string info(std::vector<iteminfo> &dump, const iteminfo_query &parts = iteminfo_query::all, int batch = 1) const;
```

Comment:
Is this really called so many times with only one parameter that you need default values for the second and third parameters? If you remove the default value for the second parameter, you don't need to include the "iteminfo_query.h" header and only need a forward declaration of `iteminfo_query`.

Reply:
Well.... I'm a C# guy - so while I can read C++ perfectly fine, and can put together _most_ things I want to do in C++, this is one of those instances where I go a bit "huh?" 😕 

Is that extra include an issue? Are there performance implications with the default values?

I simply wrote the method to have as simple as signature as possible. So while you're certainly gonna have a `dump` parameter you _might_ have a limited subset of info queried and you _might_ have a count.

What about changing it to a `iteminfo_query *fragments = nullptr` and using the current default, if no value is passed?